src/atopile/front_end.py

```diff
@@ -536,11 +417,13 @@ def _finish(self):
         with accumulate(
             errors._BaseBaseUserException, SkipPriorFailedException
         ) as ex_acc:
-            for param, (value, ctx) in self._param_assignments.items():
+            for param, (value, ctx, traceback) in self._param_assignments.items():
                 with ex_acc.collect(), ato_error_converter(), log_user_errors(logger):
                     if value is None:
                         ex = errors.UserKeyError.from_ctx(
-                            ctx, f"Parameter {param} never assigned"
+                            ctx,
+                            f"Parameter {param} never assigned",
+                            traceback=traceback,
```

Comment:
I really don't love this mode of back-tracing, so we might tinker with it further as we figure out better mechanisms